routers/web/misc/markup.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package misc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/markup"
+	"code.gitea.io/gitea/modules/markup/markdown"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/modules/web"
+
+	"mvdan.cc/xurls/v2"
+)
+
+// Render markup text in given mode
+func renderMarkup(ctx *context.Context, mode, text, urlPrefix, filePath string, wiki bool) {
```

Comment:
Look like the function is duplicated.

Reply:
I had factored out this function to deduplicate it, but then ran into circular dependencies trying to move this into `modules/markup`. I moved it to `routers/common` now, it wasn't so clear to me what the appropriate place is.